src/main/java/Main.java

```diff
@@ -1,5 +1,259 @@
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.commons.io.FileUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import static java.lang.System.*;
+
 public class Main {
+
+    static final String ANSI_RESET = "\u001B[0m";
+    static final String ANSI_BLUE = "\u001B[34m";
+    static final String ANSI_MAGENTA = "\u001b[35m";
+    private static final String ANSI_RED = "\u001b[31m";
+
+    private static JSONObject startRoom;
+    private static JSONObject endRoom;
+
+    static Map<String, Room> roomMap = new HashMap<>();
+    static Map<String, Item> itemMap = new HashMap<>();
+    private static ArrayList<Item> startInventory = new ArrayList<>();
+
+    private static String playerName;
+
+    private static void promptQuit(Player player, JSONObject originalJSON, TimeLimit timeLimit) {
+        out.println(ANSI_RED + "\nWould you like to save your place, and return later? Answer: yes / no. \n" + ANSI_RESET);
+        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+        String saveResponse;
+
+        try {
+            saveResponse = in.readLine();
+            if (saveResponse.equals("no")) {
+                out.println(ANSI_RED + "\nYou lose!\n" + ANSI_RESET);
+                exit(0);
+            } else if (saveResponse.equals("yes")) {
+
+                JSONObject copyObj = new JSONObject(originalJSON, JSONObject.getNames(originalJSON));
+                JSONArray inventoryArray = new JSONArray();
+
+                copyObj.put("current room", player.getCurrentRoom().getRoomName());
+
+                for (Item item : player.getInventory().getInventory()) {
+                    inventoryArray.put(item.getNameFromItem());
+                }
+
+                copyObj.put("player name", playerName);
+                copyObj.put("current inventory", inventoryArray);
+                copyObj.put("time limit", timeLimit.getTimeLimit() - timeLimit.getCurrentTime());
+
+                String fileName = playerName + ".json";
+
+                try (FileWriter file = new FileWriter(fileName)) {
+
+                    file.write(copyObj.toString());
+                    file.flush();
+
+                    out.println(ANSI_RED + "\nNext time you want to play, use the file 'your_name.json'. \n" + ANSI_RESET);
+                    out.println(ANSI_RED + "\nSee you later!\n" + ANSI_RESET);
+                    exit(0);
+
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            } else {
+                out.println(ANSI_RED + "\nYou need to answer 'yes' or 'no'. Try again.\n" + ANSI_RESET);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
```

Comment:
This is a long method try splitting things up a little more for organizational purposes.